app/main/views/users.py

```diff
@@ -65,32 +66,46 @@ def download_users(framework_slug):
         "application_result",
         "framework_agreement"
     ]
-    # insert header column
-    supplier_rows.insert(0, {header: header for header in supplier_headers})
 
-    def iter_csv(rows):
-
-        class Line(object):
-            def __init__(self):
-                self._line = None
-
-            def write(self, line):
-                self._line = line
-
-            def read(self):
-                return self._line
+    return Response(
+        csv_generator.iter_csv(supplier_rows, supplier_headers),
+        mimetype='text/csv',
+        headers={
+            "Content-Disposition": "attachment;filename=users-{}.csv".format(framework_slug),
+            "Content-Type": "text/csv; header=present"
+        }
+    )
 
-        line = Line()
-        writer = unicodecsv.writer(line)
-        for row in rows:
-            writer.writerow([row.get(header, '') for header in supplier_headers])
-            yield line.read()
 
+@main.route('/users/download/buyers', methods=['GET'])
+@login_required
+@role_required('admin')
+def download_buyers_and_briefs():
+    buyers_list = [user for user in data_api_client.find_users_iter(role='buyer')]
+    briefs_list = [brief for brief in data_api_client.find_briefs_iter(with_users=True)]
+
+    for buyer in buyers_list:
```

Comment:
Yeah, it could take a while! I'm not sure how to test it though with the quantity of data we have on production.

It's not going to get used very often, and only by our admin's, so i don't think it being super quick is a massive issue.

I'm going to look into how to stress test it.